### PR TITLE
bpo-33873: Fix bug in `runtest.py` and add checks for invalid `-R` parameters

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue33873

### CODE REVIEW

**Comment by pablogsal** [comment]
I have added the checks for the `--huntrleaks/-R` in `Regrtest._main` to avoid entering `Regrtest.run_test` and do the `sys.exit()` there but I am not sure what is the best location. Also, I used a new errorcode (2) but again, not sure if we want a custom error code for that.

**Comment by pablogsal** [comment]
```python
~/github/cpython bpo33873 ✔ 8s
❯ ./python -m test -R 0:0 test_os -m test_access
Invalid values for the --huntrleaks/-R parameters
                                                                                                                                               
~/github/cpython bpo33873 ✔
❯ ./python -m test -R 0:1 test_os -m test_access
Invalid values for the --huntrleaks/-R parameters
                                                                                                                                               
~/github/cpython bpo33873 ✔
❯ ./python -m test -R 1:0 test_os -m test_access
Invalid values for the --huntrleaks/-R parameters
                                                                                                                                               
~/github/cpython bpo33873 ✔
❯  ./python -m test -R 0:-2 test_list
Invalid values for the --huntrleaks/-R parameters
```

File: Lib/test/libregrtest/main.py [hunk 1]
```diff
@@ -526,6 +526,13 @@ def main(self, tests=None, **kwargs):
     def _main(self, tests, kwargs):
         self.ns = self.parse_args(kwargs)
 
+        if self.ns.huntrleaks:
+            warmup, repetitions, _ = self.ns.huntrleaks
+            if warmup < 1 or repetitions < 1:
+                print("Invalid values for the --huntrleaks/-R parameters",
```
Comment (vstinner) [inline Lib/test/libregrtest/main.py]: You must mention the minimum number of warmups and minimum number of repetitions (1 and 1).

**Review by vstinner — COMMENTED** [review]
IMHO a NEWS entry would be worth it here. You can use the name "regrtest" to name the CPython test runner libregrtest.

File: Lib/test/libregrtest/main.py [hunk 2]
```diff
@@ -526,6 +526,15 @@ def main(self, tests=None, **kwargs):
     def _main(self, tests, kwargs):
         self.ns = self.parse_args(kwargs)
 
+        if self.ns.huntrleaks:
+            warmup, repetitions, _ = self.ns.huntrleaks
+            if warmup < 1 or repetitions < 1:
+                msg = ("Invalid values for the --huntrleaks/-R parameters. The "
+                       "number of warmups and repetitions must be at least 1 "
+                       "each (1:1).")
```
Comment (vstinner) [inline Lib/test/libregrtest/main.py:534]: Nitpick: maybe write "(-R 1:1)" rather than "(1:1)"?

**Review by vstinner — APPROVED** [review]
LGTM.

Since you are now a core dev, would you like to merge it yourself? :-)

**Comment by vstinner** [comment]
"VSTS: docs — docs_20180616.05 failed"

I don't understand the failure. It says "Initialize phase: Failed" but I don't see any failure in logs. I suggest to ignore this error.

**Comment by vstinner** [comment]
@pablogsal: The change LGTM, do you want to merge it? It would help me to work on my PR #7827.

**Comment by pablogsal** [comment]
Merged! ✨

**Pull request merged**